syntax = "proto3";

package proto;

option java_multiple_files = true;
option java_package = "control.proto";

service SuggestionsService {
    rpc GetSelectedFlight(SelectedFlight) returns (SelectionResponse);
    rpc StoreOldFlight(OldFlight) returns (StoreOldResponse);
    // Logging
    rpc getLogFileSug(GetLogFileRequestSug) returns (stream GetLogFileReplySug);
}

message SelectedFlight {
    string bookingDate = 1;
    string flightDate = 2;
    string airline = 3;
    string departureAirport = 4;
    string arrivalAirport = 5;
}

message SelectionResponse {
    int32 numDaysBeforeConvenient = 1;
}

message OldFlight {
    string oldFlightsMsg = 1;
}

message StoreOldResponse {
    bool isOk = 1;
}

message GetLogFileRequestSug {
    int32 numRichiesta = 1;
}

message GetLogFileReplySug {
    bytes chunk_file  = 1;
    int32 num_chunk = 2;
}
